commands/command_migrate_import.go

```diff
@@ -29,6 +30,67 @@ func migrateImportCommand(cmd *cobra.Command, args []string) {
 	}
 	defer db.Close()
 
+	if migrateNoRewrite {
+		if len(args) == 0 {
+			ExitWithError(errors.Errorf("fatal: the migrate import command requires a list of files to import when using the --no-rewrite flag"))
+		}
+
+		ref, err := git.CurrentRef()
+		if err != nil {
+			ExitWithError(errors.Wrap(err, "fatal: unable to find current reference"))
+		}
+
+		sha, _ := hex.DecodeString(ref.Sha)
+		commit, err := db.Commit(sha)
+		if err != nil {
+			ExitWithError(errors.Wrap(err, "fatal: unable to load commit"))
+		}
+
+		root := commit.TreeID
+
+		filter := git.GetAttributeFilter(cfg.LocalWorkingDir(), cfg.LocalGitDir())
+		if len(filter.Include()) == 0 && len(filter.Exclude()) == 0 {
+			ExitWithError(errors.Errorf("fatal: no git lfs filters setup in .gitattributes"))
+		}
+
+		for _, file := range args {
```

Comment:
I think one interesting consequence of `rewriteTree()` processing one file at a time is that there are some penalties that we'll pay if we get part of the way through this loop. In particular, if there are `m` arguments where `m` is large, and the `m-1`th of them doesn't match a pattern in `.gitattributes`, we'll have created a lot of tree objects that are dangling in the object database.

So, I think that we could run this loop twice, once to check that all given filenames match at least one pattern, and then once more to call `rewriteTree()`, which would mean we spend more time here. I think that's an OK tradeoff for not creating more objects than we need.

Of course, we could also look into ways to make `rewriteTree()` take more than one file and therefore create less dangling tree objects, but I'll leave it up to you if you want to go down that path.

Reply:
Ah okay, thanks! I didn't think about the possibility of dangling objects.

I agree with you on running the loop twice and think that's the best option, since we won't be able to avoid checking files against the `.gitattributes` filters.